Library/Homebrew/diagnostic.rb

```diff
@@ -621,14 +621,15 @@ def check_for_config_scripts
         EOS
       end
 
+      def dyld_prefix; end
+
       def check_dyld_vars
```

Comment:
I'm thinking this whole function `check_dyld_vars` should be moved to `os/mac/diagnostic` and `os/linux/diagnostic, effectively duplicating it.

Reply:
can you clarify a bit? Are you suggesting renaming `check_dyld_vars` to `check_ld_vars` and using almost identical code in mac and Linux implementations?

Reply:
Yep, pretty much. Move `check_dyld_vars` to `os/mac/diagnostic.rb`, and create a new function `check_ld_vars` in `os/linux/diagnostic.rb` that is essentially a duplicate of `check_dyld_vars`.